hikari/impl/cache.py

```diff
@@ -278,19 +290,28 @@ def _get_emojis_view(  # TODO: split out the two cases (specific guild vs global
             if emoji_data.user_id is not None:
                 cached_users[emoji_data.user_id] = self._user_entries[emoji_data.user_id]
 
-        return cache_utility.StatefulCacheMappingView(
+        return cache_utility.CacheMappingView(
             cached_emojis, builder=lambda emoji: self._build_emoji(emoji, cached_users=cached_users)
         )
 
     def get_emojis_view(self) -> cache.CacheView[snowflakes.Snowflake, emojis.KnownCustomEmoji]:
+        if not self._is_cache_enabled_for("emojis"):
+            return cache_utility.EmptyCacheView()
+
         return self._get_emojis_view()
 
     def get_emojis_view_for_guild(
         self, guild_id: snowflakes.Snowflake, /
     ) -> cache.CacheView[snowflakes.Snowflake, emojis.KnownCustomEmoji]:
+        if not self._is_cache_enabled_for("emojis"):
+            return cache_utility.EmptyCacheView()
+
         return self._get_emojis_view(guild_id=guild_id)
 
     def set_emoji(self, emoji: emojis.KnownCustomEmoji, /) -> None:
+        if not self._is_cache_enabled_for("emojis"):
```

Comment:
might as well make constants for these names

Reply:
Was thinking about that, will do 👍